beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go

```diff
@@ -231,30 +234,37 @@ func (vs *Server) BuildBlockParallel(ctx context.Context, sBlk interfaces.Signed
 		vs.setBlsToExecData(sBlk, head)
 	}()
 
-	localPayload, overrideBuilder, err := vs.getLocalPayload(ctx, sBlk.Block(), head)
-	if err != nil {
-		return status.Errorf(codes.Internal, "Could not get local payload: %v", err)
-	}
-
-	// There's no reason to try to get a builder bid if local override is true.
-	var builderPayload interfaces.ExecutionData
-	var builderKzgCommitments [][]byte
-	overrideBuilder = overrideBuilder || skipMevBoost // Skip using mev-boost if requested by the caller.
-	if !overrideBuilder {
-		builderPayload, builderKzgCommitments, err = vs.getBuilderPayloadAndBlobs(ctx, sBlk.Block().Slot(), sBlk.Block().ProposerIndex())
+	blockEpoch := slots.ToEpoch(sBlk.Block().Slot())
+	winningBid := primitives.ZeroWei
+	var bundle *enginev1.BlobsBundle
+	if blockEpoch >= params.BeaconConfig().BellatrixForkEpoch {
+		local, err := vs.getLocalPayload(ctx, sBlk.Block(), head)
 		if err != nil {
-			builderGetPayloadMissCount.Inc()
-			log.WithError(err).Error("Could not get builder payload")
+			return primitives.ZeroWei, nil, status.Errorf(codes.Internal, "Could not get local payload: %v", err)
 		}
-	}
 
-	if err := setExecutionData(ctx, sBlk, localPayload, builderPayload, builderKzgCommitments, builderBoostFactor); err != nil {
-		return status.Errorf(codes.Internal, "Could not set execution data: %v", err)
+		// There's no reason to try to get a builder bid if local override is true.
+		var builderBid builderapi.Bid
+		if !(local.OverrideBuilder || skipMevBoost) {
+			builderBid, err = vs.getBuilderPayloadAndBlobs(ctx, sBlk.Block().Slot(), sBlk.Block().ProposerIndex())
+			if err != nil {
+				builderGetPayloadMissCount.Inc()
+				log.WithError(err).Error("Could not get builder payload")
+			} else if builderBid == nil {
+				builderGetPayloadMissCount.Inc()
+				log.WithError(err).Error("Could not get builder payload")
```

Comment:
This is wonky, I actually don't think we need to do anything `if err == nil && builderBid == nil` - this means `getBuilderPayloadAndBlobs` decided we didn't need to call the builder, because it's disabled or pre-bellatrix.